stories/assets/js/select.js

```diff
@@ -1,153 +1,24 @@
-  }
-
-
-  close() {
-    this.$selectOptions.removeClass(this.activeClass);
-    this.$currentSelect.removeClass(this.expandedClass);
-  }
-
-  setupFocus() {
-    if (this.optionSelectedId) {
-      return;
-    }
-
-    this.focusFirstItem();
-  }
-
-  focusFirstItem() {
-    const firstItem = this.$selectOptions[0].querySelector('[role="option"]');
-
-    if (firstItem) {
-      this.focusItem(firstItem);
-    }
-  }
-
-  focusItem(element) {
-    this.defocusItem(document.getElementById(this.optionSelectedId));
-    this.$selectOptions.attr('aria-activedescendant', element.id);
-    this.optionSelectedId = element.id;
-    this.value = element.dataset.value;
-
-    if (this.$selectOptions[0].scrollHeight > this.$selectOptions[0].clientHeight) {
-      const scrollBottom = this.$selectOptions[0].clientHeight + this.$selectOptions[0].scrollTop;
-      const elementBottom = element.offsetTop + element.offsetHeight;
-      if (elementBottom > scrollBottom) {
-        this.$selectOptions[0].scrollTop = elementBottom - this.$selectOptions[0].clientHeight;
-      } else if (element.offsetTop < this.$selectOptions[0].scrollTop) {
-        this.$selectOptions[0].scrollTop = element.offsetTop;
-      }
-    }
-
-    this.changeSelectedOption();
-  }
-
-  defocusItem(element) {
-    if (!element) {
-      return;
-    }
-  }
-
-  checkKeyPress(evt) {
-    const key = evt.which || evt.keyCode;
-    let nextItem = document.getElementById(this.optionSelectedId);
-
-    switch (key) {
-      case window.UNDP.keyCode.UP:
-      case window.UNDP.keyCode.DOWN:
-        evt.preventDefault();
-
-        if (key === window.UNDP.keyCode.UP) {
-          nextItem = nextItem.previousElementSibling;
-        } else {
-          nextItem = nextItem.nextElementSibling;
-        }
-
-        if (nextItem) {
-          this.focusItem(nextItem);
-        }
-
-        break;
-      case window.UNDP.keyCode.SPACE:
-      case window.UNDP.keyCode.RETURN:
-        evt.preventDefault();
-        this.value = nextItem.dataset.value;
-
-        this.close();
-        this.changeSelectedOption();
-        break;
-      default:
-        break;
-    }
-  }
-}
-
 export function select() {
-  $('.select-box li').on('click', function(){
+  let selectButton = $('.select-box button');
+  let selectList = $('.select-box li');
+
+  selectButton.on('click', function(){
+    $(this).parent().toggleClass('expanded').find('ul').toggleClass('active');
+  });
+  selectList.on('click keypress', function(){
     $(this).parent().siblings().text($(this).find('span').text());
   });
-  const selects = $('[data-select]');
-  selects.each((i, select) => {
-    const selectInstance = new Select(select);
-    selectInstance.init();
+  $(document).mouseup(function(e) {
+    if (!selectButton.is(e.target) && selectButton.has(e.target).length === 0) 
+    {
+      selectButton.parent().removeClass('expanded').find('ul').removeClass('active');
+    }
+  });
+  selectList.on('keypress',function(e) {
```

Comment:
@Ashish-srij Can you include the `selectList.on()` method with the `on()` method used above rather than writing it twice? Something like:
```
selectList.on('click keypress', function() {
  if (e.type == 'keypress' && e.which == 13) {
    $(this).parent().removeClass('active').parents().removeClass('expanded');
  }
  $(this).parent().siblings().text($(this).find('span').text());
});
```

Reply:
@shishirphadke After remove undp.js we don't need the if condition here so I removed it. just add click and keypress in single function.